OsmAnd/src/net/osmand/plus/render/MapRenderRepositories.java

```diff
@@ -303,8 +306,11 @@ public void checkInitialized(final int zoom, NativeOsmandLibrary library, int le
 		boolean useLive = context.getSettings().USE_OSM_LIVE_FOR_ROUTING.get();
 		for (String mapName : files.keySet()) {
 			BinaryMapIndexReader fr = files.get(mapName);
-			if (fr != null && (fr.containsMapData(leftX, topY, rightX, bottomY, zoom) || 
+			if (fr != null && (fr.containsMapData(leftX, topY, rightX, bottomY, zoom) ||
 					fr.containsRouteData(leftX, topY, rightX, bottomY, zoom))) {
+				if (!Algorithms.isEmpty(fr.getCountryName())) {
```

Comment:
what if few files will contain map data and first one is japan but last one is not japan?

Reply:
fixed